flink-connectors/flink-orc/src/main/java/org/apache/flink/orc/OrcFileWriter.java

```diff
@@ -0,0 +1,269 @@
+	}
+
+	private TableSchema orcSchemaToTableSchema(TypeDescription orcSchema) {
+		List<String> fieldNames = orcSchema.getFieldNames();
+		List<TypeDescription> typeDescriptions = orcSchema.getChildren();
+		List<TypeInformation> typeInformations = new ArrayList<>();
+
+		typeDescriptions.forEach(typeDescription -> {
+			typeInformations.add(schemaToTypeInfo(typeDescription));
+		});
+
+		return new TableSchema(
+			fieldNames.toArray(new String[fieldNames.size()]),
+			typeInformations.toArray(new TypeInformation[typeInformations.size()]));
+	}
+
+	@Override
+	public void write(T element) throws IOException {
+		Boolean isFill = orcBatchWriter.fill(rowBatch, element);
+		if (!isFill) {
+			writer.addRowBatch(rowBatch);
+			writedRowSize += writedRowSize + rowBatch.size;
+			rowBatch.reset();
+		}
+	}
+
+	@Override
+	public long flush() throws IOException {
+		writer.addRowBatch(rowBatch);
+		writedRowSize += rowBatch.size;
+		rowBatch.reset();
+		return writedRowSize;
+	}
+
+	@Override
+	public long getPos() throws IOException {
+		return writedRowSize;
+	}
+
+	@Override
+	public Writer<T> duplicate() {
+		return new OrcFileWriter<>(metaSchema, compressionKind);
+	}
+
+	@Override
+	public void close() throws IOException {
+		flush();
+		if (rowBatch.size != 0) {
+			writer.addRowBatch(rowBatch);
+			rowBatch.reset();
+		}
+		writer.close();
+		super.close();
+	}
+
+	@Override
+	public int hashCode() {
+		return Objects.hash(super.hashCode(), schema, metaSchema, rowBatch, orcBatchWriter);
+	}
+
+	@Override
+	public boolean equals(Object other) {
+		if (this == other) {
+			return true;
+		}
+		if (other == null) {
+			return false;
+		}
+		if (getClass() != other.getClass()) {
+			return false;
+		}
+		OrcFileWriter<T> writer = (OrcFileWriter<T>) other;
+		return Objects.equals(schema, writer.schema)
+			&& Objects.equals(metaSchema, writer.metaSchema)
+			&& Objects.equals(rowBatch, writer.rowBatch)
+			&& Objects.equals(orcBatchWriter, writer.orcBatchWriter);
+	}
+
+	/**
+	 * The writer that fill the records into the batch.
+	 */
+	private class OrcBatchWriter {
+
+		private List<TypeInformation> typeInfos;
+
+		public OrcBatchWriter(List<TypeInformation> typeInfos) {
+			this.typeInfos = typeInfos;
+		}
+
+		/**
+		 * Fill the record into {@code VectorizedRowBatch}, return false if batch is full.
+		 *
+		 * @param batch  An reusable orc VectorizedRowBatch.
+		 * @param record Input record.
+		 * @return Return false if batch is full.
+		 */
+		private boolean fill(VectorizedRowBatch batch, T record) {
+			// If the batch is full, write it out and start over.
+			if (batch.size == batch.getMaxSize()) {
+				return false;
+			} else {
+				IntStream.range(0, typeInfos.size()).forEach(
+					index -> {
+						setColumnVectorValueByType(typeInfos.get(index), batch, index, batch.size, record);
+					}
+				);
+				batch.size += 1;
+				return true;
+			}
+		}
+
+		private void setColumnVectorValueByType(TypeInformation typeInfo,
```

Comment:
How do you handle NULL values?

Reply:
I will verify the NULL situation and ping you again. @xndai Thank you ~

Reply:
With current implementation, I think ```T record``` would cause NULL value.